src/librustc/hir/lowering/expr.rs

```diff
@@ -658,8 +651,24 @@ impl LoweringContext<'_> {
             attrs: ThinVec::new(),
         });
 
+        // mut pinned => loop { ... }
+        let pinned_arm = self.arm(hir_vec![pinned_pat], loop_expr);
+
+        // match <expr> {
+        //     mut pinned => loop { .. }
+        // }
+        let outer_match_expr = {
+            let expr = P(self.lower_expr(expr));
+            P(self.expr_match(
+                span,
+                expr,
+                hir_vec![pinned_arm],
+                hir::MatchSource::AwaitDesugar,
+            ))
+        };
+
         hir::ExprKind::Block(
```

Comment:
I don't think that there's a need to wrap the match expression in a block now.

Reply:
Fixed, thanks!

Reply:
The comment for the function also needs updating

Reply:
Oops, fixed.